corehq/apps/reports/standard/tableau.py

```diff
@@ -12,8 +12,10 @@
 from corehq.apps.reports.models import TableauVisualization
 from corehq.apps.domain.decorators import login_and_domain_required
 from corehq.apps.domain.views.base import BaseDomainView
+from corehq.apps.locations.permissions import location_safe
 
 
+@location_safe
```

Comment:
You may also be able to use `@conditionally_location_safe` for these, and then check that the viz is marked as safe in the fn you pass to it